torchtext/utils.py

```diff
@@ -27,39 +28,56 @@ def inner(b=1, bsize=1, tsize=None):
     return inner
 
 
-def download_from_url(url, path):
-    """Download file, with logic (from tensor2tensor) for Google Drive
+def download_from_url(url, root='.data', filename=None, overwrite=False):
+    """Download file, with logic (from tensor2tensor) for Google Drive.
+    Returns the path to the downloaded file.
 
     Arguments:
-        url: the url for online Dataset
-        path: directory and filename for the downloaded dataset.
+        url: the url of the file
+        root: download folder used to store the file in (.data)
+        filename: explicitly set the filename, otherwise attempts to
+            detect the file name from URL header. (None)
+        overwrite: overwrite existing files (False)
 
     Examples:
         >>> url = 'http://www.quest.dcs.shef.ac.uk/wmt16_files_mmt/validation.tar.gz'
-        >>> path = './validation.tar.gz'
-        >>> torchtext.utils.download_from_url(url, path)
+        >>> torchtext.utils.download_from_url(url)
+        >>> '.data/validation.tar.gz'
     """
 
-    def process_response(r):
+    def process_response(r, root, filename):
```

Comment:
So this changes the API. Make sure it doesn't break any existing applications.

Reply:
Ah, but it's a local function. We should prepend it with "_"